### PR TITLE
tls: move out of extensions

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: low: code move
Testing: n/a
Docs Changes: n/a
Release Notes: n/a
https://github.com/envoyproxy/envoy/issues/31113

### CODE REVIEW

**Comment by alyssawilk** [comment]
@phlax is there a way to exempt testdata from glint checks?

CodeChecker ERROR [glint] Missing final newline: test/common/tls/test_data/aes_128_key
CodeChecker ERROR [glint] Missing final newline: test/common/tls/test_data/ticket_key_a
CodeChecker ERROR [glint] Missing final newline: test/common/tls/test_data/ticket_key_b
CodeChecker ERROR [glint] Missing final newline: test/common/tls/test_data/ticket_key_wrong_len
CodeChecker ERROR [glint] Check failed
CodeChecker ERROR ERRORS Summary [glint]:
--------------------------------------------

**Comment by phlax** [comment]
yep see here https://github.com/envoyproxy/envoy/blob/09042c3a24ece2fae5b88f91e8af889a3feab848/.gitattributes#L4-L5

**Comment by alyssawilk** [comment]
oh perfect, thanks.

**Comment by alyssawilk** [comment]
Ok I think this is finally passing, modulo windows which has been having CI issues.

File: ci/osx-build-config/extensions_build_config.bzl [hunk 1]
```diff
@@ -16,7 +16,6 @@ EXTENSIONS = {
     "envoy.network.dns_resolver.getaddrinfo": "//source/extensions/network/dns_resolver/getaddrinfo:config",
     "envoy.transport_sockets.http_11_proxy": "//source/extensions/transport_sockets/http_11_proxy:upstream_config",
     "envoy.transport_sockets.raw_buffer": "//source/extensions/transport_sockets/raw_buffer:config",
-    "envoy.transport_sockets.tls": "//source/extensions/transport_sockets/tls:config",
```
Comment (ggreenway) [inline ci/osx-build-config/extensions_build_config.bzl:19]: Following the pattern of other core/built-in things used at extension points, such as the tcp_proxy network filter and the raw_buffer transport socket, I think there should still be a stub-extension that can be enabled/disabled here, and just the factory code live in `source/extensions`

File: source/exe/BUILD [hunk 2]
```diff
@@ -144,6 +144,8 @@ envoy_cc_library(
         ":main_common_with_all_extensions_lib",
         # These are compiled as extensions so Envoy Mobile doesn't have to link them in.
         # Envoy requires them.
+        "//source/common/tls/cert_validator:cert_validator_lib",
```
Comment (ggreenway) [inline source/exe/BUILD:147]: these changes can go away when it has a place-holder extension

**Review by ggreenway — CHANGES_REQUESTED** [review]
/wait

File: source/extensions/transport_sockets/tap/BUILD [hunk 3]
```diff
@@ -52,6 +52,10 @@ envoy_cc_extension(
     name = "config",
     srcs = envoy_select_admin_functionality(["config.cc"]),
     hdrs = envoy_select_admin_functionality(["config.h"]),
+    extra_visibility = [
```
Comment (ggreenway) [inline source/extensions/transport_sockets/tap/BUILD:55]: Why is this needed?

**Review by ggreenway — APPROVED** [review]
LGTM! I did my best to review, but it's possible some touch-ups will be needed.

**Pull request merged**